plugins/CoreConsole/Commands/DevelopmentSyncUITestScreenshots.php

```diff
@@ -73,42 +71,166 @@ protected function execute(InputInterface $input, OutputInterface $output)
-        $output->writeln('');
-        $commands = "cd tests/UI/expected-ui-screenshots
+        $downloadToPath = $this->getDownloadToPath($input);
+        $commands = "
+cd $downloadToPath
 git pull
 git add .
-git commit -m '' # WRITE A COMMIT MESSAGE
-git push
+git commit -m '' # Write a good commit message
+git push";
+
+        $plugin = $input->getOption('plugin');
+        if(empty($plugin)) {
+            $commands .= "
 cd ..
 git pull
-git add expected-ui-screenshots
-git commit -m '' #WRITE A COMMIT MESSAGE
-git push";
+git add expected-ui-screenshots/
+git commit -m '' # Copy paste the good commit message
+git push
+cd ../../\n\n";
+        } else {
+            $commands .= "
+cd ../../../../../\n\n";
+        }
         $output->writeln($commands);
     }
+
+    protected function getUrlBase(InputInterface $input)
+    {
+        $buildNumber = $input->getArgument('buildnumber');
+        if (empty($buildNumber)) {
+            throw new \InvalidArgumentException('Missing build number.');
+        }
+
+        $plugin = $input->getOption('plugin');
+        if ($plugin) {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master.%s/%s', $plugin, $buildNumber);
+        } else {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master/%s', $buildNumber);
+        }
+        return $urlBase;
+    }
+
+    protected function getDownloadToPath(InputInterface $input)
+    {
+        $plugin = $input->getOption('plugin');
+
+        $downloadTo = PIWIK_DOCUMENT_ROOT . "/";
+        if (empty($plugin)) {
+            $downloadTo .= "tests/UI/expected-ui-screenshots/";
+        } else {
+            $downloadTo .= "plugins/$plugin/tests/UI/expected-ui-screenshots/";
+        }
+        if(is_dir($downloadTo)) {
+            return $downloadTo;
+        }
+        $downloadTo = str_replace("tests/", "Test/", $downloadTo);
+        if(is_dir($downloadTo)) {
+            return $downloadTo;
+        }
+
+        throw new \Exception("Download to path could not be found: $downloadTo");
+    }
+
+    protected function getDiffviewerContent(InputInterface $input)
+    {
+        $this->urlBase = $this->getUrlBase($input);
+        $diffviewerUrl = $this->getDiffviewerUrl($this->urlBase);
+
+        $diffviewer = $this->downloadDiffviewer($diffviewerUrl);
+        if($diffviewer) {
+            return $diffviewer;
+        }
+
+        // Maybe this is a Premium Piwik PRO plugin...
+        return $this->getDiffviewContentForPrivatePlugin($input);
+    }
+
+    protected function getDiffviewContentForPrivatePlugin(InputInterface $input)
+    {
+        $httpUser = $input->getOption('http-user');
+        $httpPassword = $input->getOption('http-password');
+        if (empty($httpUser) || empty($httpPassword)) {
+            Log::info("--http-user and --http-password was not specified, skip download of private plugins screenshots.");
```

Comment:
You should, I think, not use `Log` since it's been deprecated for a while now. What I've been doing is storing the PSR $logger in a member and initializing upon construction:

```
    public function __construct(LoggerInterface $logger = null)
    {
        $this->logger = $logger ?: StaticContainer::get('Psr\Log\LoggerInterface');
    }
```

I think it is also acceptable to access it directly via `StaticContainer::get('Psr\Log\LoggerInterface')` (CC @mnapoli is this right?)

You also might want to use OutputInterface directly, instead of using Log, so the output is clearer. But this isn't that important.